libecole/src/scip/scimpl.cpp

```diff
@@ -100,6 +132,37 @@ void scip::Scimpl::solve_iter_branch(SCIP_RESULT result) {
 		*final_result = result;
 		return SCIP_OKAY;
 	});
+
+	m_controller->wait_thread();
+}
+
+void Scimpl::solve_iter_start_primalsearch(
+	int trials_per_node,
+	int depth_freq,
+	int depth_start,
+	int depth_stop,
+	SCIP_HEUR** heur) {
+	auto* const scip_ptr = get_scip_ptr();
+	m_controller = std::make_unique<utility::Controller>([=](std::weak_ptr<utility::Controller::Executor> weak_executor) {
+		scip::call(
+			SCIPincludeObjHeur,
+			scip_ptr,
+			new ReverseHeur(
+				scip_ptr, std::move(weak_executor), trials_per_node, depth_freq, depth_start, depth_stop),  // NOLINT
+			true);
+		scip::call(SCIPsolve, scip_ptr);  // NOLINT
+	});
+
+	m_controller->wait_thread();
+	*heur = SCIPfindHeur(scip_ptr, scip::ReverseHeur::name);  // can we not rely on the name ?
```

Comment:
Unfortunately it seems we cannot.
```suggestion
	*heur = SCIPfindHeur(scip_ptr, scip::ReverseHeur::name);
```

Reply:
What I had in mind was maybe to recover `heur` not here but instead when SCIP calls the heuristic, i.e, get it from the argument in `ReverseHeur::scip_exec()` and pass it forward to `action_func()` somehow. But I suppose it will require a bit of code refactor to allow that.